Source/Core/Core/NetworkCaptureLogger.h

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <memory>
+
+#ifdef _WIN32
+#include <WinSock2.h>
+using socklen_t = int;
+#else
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <sys/types.h>
+#endif
+
+#include "Common/CommonTypes.h"
+
+namespace Common
+{
+class PCAP;
+}
+
+namespace Core
+{
+enum class NetworkCaptureType
+{
+  None,
+  Raw,
+  PCAP,
+};
+
+class NetworkCaptureLogger
+{
+public:
+  virtual ~NetworkCaptureLogger();
+
+  virtual void LogSSLRead(const void* data, std::size_t length, s32 socket) = 0;
+  virtual void LogSSLWrite(const void* data, std::size_t length, s32 socket) = 0;
+
+  virtual void LogRead(const void* data, std::size_t length, s32 socket, sockaddr* from) = 0;
+  virtual void LogWrite(const void* data, std::size_t length, s32 socket, sockaddr* to) = 0;
+
+  virtual NetworkCaptureType GetCaptureType() const = 0;
+};
+
+class DummyNetworkCaptureLogger : public NetworkCaptureLogger
```

Comment:
These loggers don't seem to be used anywhere?

But they could potentially be marked `final`